genesyscloud/workforcemanagement_businessunits/resource_genesyscloud_workforcemanagement_businessunits.go

```diff
@@ -0,0 +1,130 @@
+*/
+
+// getAllAuthWorkforceManagementBusinessUnits retrieves all the workforce management business units via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthWorkforceManagementBusinessUnits(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := newWorkforceManagementBusinessUnitsProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	businessUnitResponses, resp, err := proxy.getAllWorkforceManagementBusinessUnits(ctx)
+	if err != nil {
+		return nil, util.BuildAPIDiagnosticError(ResourceName, fmt.Sprintf("Failed to get workforce management business units: %v", err), resp)
+	}
+
+	for _, businessUnitResponse := range *businessUnitResponses {
+		resources[*businessUnitResponse.Id] = &resourceExporter.ResourceMeta{BlockLabel: *businessUnitResponse.Name}
+	}
+
+	return resources, nil
+}
+
+// createWorkforceManagementBusinessUnit is used by the workforcemanagement_businessunits resource to create Genesys cloud workforce management business unit
+func createWorkforceManagementBusinessUnit(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getWorkforceManagementBusinessUnitsProxy(sdkConfig)
+
+	createBusinessUnitRequest := getCreateWorkforcemanagementBusinessUnitRequestFromResourceData(d)
+
+	log.Printf("Creating workforce management business unit %s", *createBusinessUnitRequest.Name)
+	businessUnitResponse, resp, err := proxy.createWorkforceManagementBusinessUnit(ctx, &createBusinessUnitRequest)
+	if err != nil {
+		return util.BuildAPIDiagnosticError(ResourceName, fmt.Sprintf("Failed to create workforce management business unit: %s", err), resp)
+	}
+
+	d.SetId(*businessUnitResponse.Id)
+	log.Printf("Created workforce management business unit %s", *businessUnitResponse.Id)
+	return readWorkforceManagementBusinessUnit(ctx, d, meta)
+}
+
+// readWorkforceManagementBusinessUnit is used by the workforcemanagement_businessunits resource to read a workforce management business unit from genesys cloud
+func readWorkforceManagementBusinessUnit(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getWorkforceManagementBusinessUnitsProxy(sdkConfig)
+	cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceWorkforcemanagementBusinessunits(), constants.ConsistencyChecks(), ResourceName)
+
+	log.Printf("Reading workforce management business unit %s", d.Id())
+
+		return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+			businessUnitResponse, resp, getErr := proxy.getWorkforceManagementBusinessUnitById(ctx, d.Id())
+			if getErr != nil {
+				// Retry on 409 (conflict) errors
+				if util.IsStatus409(resp) {
```

Comment:
Just going to drop a note in general that 409s in many cases are non-retryable errors. Specifically, if a version number is supplied and you get a 409, it will fail forever going forward no matter how many times you retry. Now if a version number isn't supplied, then a 409 may or may not work. So more pointing that out for general awareness. But in this case, it seems super weird to handle a 409 on a GET only operation (this looks to be a BU get) when that's not even something like the backend can throw from my knowledge. So what was the motivation specifically calling this out here?